testkit/src/main/java/org/apache/calcite/util/Smalls.java

```diff
@@ -1492,4 +1492,24 @@ public static ByteString eval(String s) {
       }
     }
   }
+
+  /** User-defined function that convert Character[] to VARCHAR. */
+  public static class CharacterArrayFunction {
```

Comment:
I am not sure I understand this comment; Character[] is a Java type, while VARCHAR is a SQL type. 

Is there a test for the BINARY case which is described in the issue title?

Reply:
I have changed the title of the jira and pr. I recorded the problem with the byte[] type in the new jira https://issues.apache.org/jira/browse/CALCITE-7187. Here I changed the comment content to: User-defined function with return type Character[].